src/lib/homestore/heap_chunk_selector.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <homestore/chunk_selector.h>
+#include <homestore/vchunk.h>
+#include <homestore/homestore_decl.hpp>
+
+#include <queue>
+#include <vector>
+#include <mutex>
+#include <utility>
+#include <memory>
+#include <functional>
+#include <atomic>
+
+#include <folly/AtomicHashMap.h>
+
+namespace homeobject {
+
+using csharedChunk = homestore::cshared<homestore::Chunk>;  
+
+class HeapChunkSelector : public homestore::ChunkSelector {
+public:
+    HeapChunkSelector();
+    HeapChunkSelector(const uint32_t&, const uint32_t&, const uint32_t&);
+    ~HeapChunkSelector() = default;
+
+    using VChunk = homestore::VChunk;
+    class VChunkComparator {
+    public:
+        bool operator()(VChunk& lhs, VChunk& rhs) {
+            return lhs.available_blks() < rhs.available_blks();
+        }
+    };
+
+    using VChunkHeap = std::priority_queue<VChunk, std::vector<VChunk>, VChunkComparator>;
+
+    void add_chunk(csharedChunk&) override;
+    void foreach_chunks(std::function< void(csharedChunk&) >&& cb) override;
+    csharedChunk select_chunk([[maybe_unused]]homestore::blk_count_t nblks, const homestore::blk_alloc_hints& hints);
+
+    // this function is used to return a chunk back to ChunkSelector when sealing a shard, and will only be used by Homeobject.
+    void release_chunk(const uint32_t&);
```

Comment:
Just curious do we need to const pass-by-reference for uint32_t or easily be `void release_chunk(uint32_t)`. Isn't pass-by-value for this is more efficient and also simpler?

Reply:
Yes; reference are really `pass by pointer` anyways; which on a 64-bit platform is a "technically" more costly than a uint32_t.

Reply:
yes that is right